src/components/income/OtherInvestments.tsx

```diff
@@ -0,0 +1,198 @@
+
+interface Show<A> {
+  (a: A): string
+}
+
+const toShow = <A,>(f: (a: A) => string): Show<A> => f
+
+const show =
+  <A,>(shows: Show<A>) =>
+  (a: A): string =>
+    shows(a)
+
+const showAssetType: Show<AssetType> = toShow((p) => {
+  switch (p) {
+    case 'Security':
+      return 'Security (Stock, bond, option, mutual fund, etc.)'
+    case 'Real Estate':
+      return 'Real Estate'
+  }
+})
+
+interface AssetUserInput {
+  name: string
+  positionType: AssetType
+  openDate?: Date
+  closeDate?: Date
+  openPrice: string
+  closePrice: string
+  quantity: string
+  state?: State
+}
+
+const blankUserInput: AssetUserInput = {
+  name: '',
+  positionType: 'Security',
+  openPrice: '',
+  closePrice: '',
+  quantity: ''
+}
+
+const toUserInput = (f: Asset<Date>): AssetUserInput => ({
+  ...blankUserInput,
+  ...f,
+  openPrice: f.openPrice.toString(),
+  closePrice: f.closePrice?.toString() ?? '',
+  quantity: f.quantity.toString()
+})
+
+const toAsset = (input: AssetUserInput): Asset<Date> | undefined => {
+  const { name, openDate, closeDate, openPrice, closePrice, quantity, state } =
+    input
+  if (name === '' || openDate === undefined) {
+    return undefined
+  }
+  return {
+    name,
+    openDate,
+    closeDate,
+    openPrice: Number(openPrice),
+    closePrice: Number(closePrice),
+    quantity: Number(quantity),
+    state
+  }
+}
+
+export default function OtherInvestments(): ReactElement {
+  const assets = useSelector((state: YearsTaxesState) => state.assets)
+
+  const methods = useForm<AssetUserInput>()
+  const { handleSubmit, watch } = methods
+  const positionType = watch('positionType')
+  const dispatch = useDispatch()
+
+  const { onAdvance, navButtons } = usePager()
+
+  const onSubmitAdd = (formData: AssetUserInput): void => {
+    const payload = toAsset(formData)
+    if (payload !== undefined) {
+      dispatch(addAsset(payload))
+    }
+  }
+
+  const onSubmitEdit =
+    (index: number) =>
+    (formData: AssetUserInput): void => {
+      const payload = toAsset(formData)
+      if (payload !== undefined) {
+        dispatch(editAsset({ value: payload, index }))
+      }
+    }
+
+  const form: ReactElement | undefined = (
+    <FormListContainer<AssetUserInput>
+      onSubmitAdd={onSubmitAdd}
+      onSubmitEdit={onSubmitEdit}
+      items={assets.map((a) => toUserInput(a))}
+      removeItem={(i) => dispatch(removeAsset(i))}
+      primary={(f) => f.name}
+      secondary={(f) =>
+        `${f.openDate?.toDateString() ?? ''}: ${f.quantity} @ $${f.openPrice}`
+      }
+    >
+      {' '}
+      <Grid container spacing={2}>
+        <p>Manage Assets</p>
+        <GenericLabeledDropdown<AssetType>
+          label="Asset Type"
+          name="positionType"
+          dropDownData={['Security', 'Real Estate']}
+          keyMapping={(x) => x}
+          textMapping={show(showAssetType)}
+          valueMapping={(x) => x}
+        />
+        <LabeledInput
+          label="Address"
```

Comment:
"Address" only makes sense for Real Estate. This should just be "Description" or something similar for a Security.

Reply:
Ah yes, meant to add a switch there.